src/ring_theory/polynomial/hermite.lean

```diff
@@ -125,4 +129,32 @@ end
 
 end coeff
 
+/-! ### Lemmas about `hermite n` and gaussians -/
+section gaussian
+
+lemma hermite_gauss_succ (n : ℕ) (x : ℝ) :
+  (-1 : ℝ)^(n+1) * real.exp (x^2 / 2) * (deriv^[n+1] (λ y, real.exp (-(y^2 / 2))) x)
+  = x * ((-1 : ℝ)^n * real.exp (x^2 / 2) * (deriv^[n] (λ y, real.exp (-(y^2 / 2))) x)) -
+    deriv (λ x, (-1 : ℝ)^n * real.exp (x^2 / 2) * (deriv^[n] (λ y, real.exp (-(y^2 / 2))) x)) x :=
+begin
+  rw [function.iterate_succ_apply', deriv_mul, deriv_const_mul,
+      (by simp [mul_comm] : deriv (λ y, real.exp (y^2 / 2)) x = x * real.exp (x^2 / 2))],
+  ring_exp,
+  { simp },
+  { simp },
+  { apply (cont_diff_top_iff_deriv.mp (cont_diff.iterate_deriv _ _ _)).1,
+    exact ((cont_diff_id.pow 2).div_const 2).neg.exp },
+end
+
+/- `hermite n` is (up to sign) the factor appearing in `deriv^[n]` of a gaussian -/
+lemma hermite_eq_deriv_gaussian : ∀ (x : ℝ) (n : ℕ),
+  aeval x (hermite n) = (-1 : ℝ)^n * real.exp (x^2 / 2) * (deriv^[n] (λ y, real.exp (-(y^2 / 2))) x)
```

Comment:
What are you thoughts on instead writing this (and the lemma above) as
```suggestion
  aeval x (hermite n) = (-1 : ℝ)^n * (deriv^[n] (λ y, real.exp (-(y^2 / 2))) x) / real.exp (-(x^2 / 2))
```
This is a bit more directly "the factor", since the former `gaussian` appears in the expression twice.

Reply:
No strong preference, except that using `real.div` means more steps, since we will have to start reminding Lean that `real.exp x \ne 0` and we also probably don't want the derivative rules to use `deriv.div`. It's also not clear to me whether we would rather use the form `a / b` or `b⁻¹ * a`.

My inclination would be to leave it as-is.

Reply:
Please do move the arguments before the colon, in any case.

Reply:
A version of this lemma if you prefer:
```lean
/- `hermite n` is (up to sign) the factor appearing in `deriv^[n]` of a gaussian -/
lemma hermite_eq_deriv_gaussian' : ∀ (n : ℕ) (x : ℝ),
  aeval x (hermite n) = 
  (-1 : ℝ)^n * (deriv^[n] (λ y, real.exp (-(y^2 / 2))) x) / real.exp (-(x^2 / 2))
| 0       _  := by simp [← real.exp_sub]
| (m + 1) x :=
begin
  have : ∀ x n,
    (-1 : ℝ)^(n+1) * (deriv^[n+1] (λ y, real.exp (-(y^2 / 2))) x) / real.exp (-(x^2 / 2))
    = x * ((-1 : ℝ)^n * (deriv^[n] (λ y, real.exp (-(y^2 / 2))) x)) / real.exp (-(x^2 / 2)) -
      deriv (λ z, (-1 : ℝ)^n * (deriv^[n] (λ y, real.exp (-(y^2 / 2))) z) / real.exp (-(z^2 / 2))) x,
  { intros x n,
    simp_rw [real.exp_neg, div_inv_eq_mul, mul_assoc, ← real.exp_neg, 
             mul_comm _ (real.exp _), ← mul_assoc],
    rw [function.iterate_succ_apply', deriv_mul, deriv_const_mul,
        (by simp [mul_comm] : deriv (λ y, real.exp (y^2 / 2)) x = x * real.exp (x^2 / 2))],
    ring_exp,
    { simp },
    { simp },
    { apply (cont_diff_top_iff_deriv.mp (cont_diff.iterate_deriv _ _ _)).1,
      exact ((cont_diff_id.pow 2).div_const 2).neg.exp } },
  simp_rw [this, hermite_succ, ← mul_div x, ← hermite_eq_deriv_gaussian'],
  simp [aeval_def, eval₂_eq_eval_map],
end
```

Reply:
I think the key advantage of the `/` spelling is that you can add `let gaussian :=` inside the proof to make it more readable.

Reply:
I see. Here's what I come up with then:
```lean
/- `hermite n` is (up to sign) the factor appearing in `deriv^[n]` of a gaussian -/
lemma hermite_eq_deriv_gaussian' : ∀ (n : ℕ) (x : ℝ),
  aeval x (hermite n) =
  (-1 : ℝ)^n * (deriv^[n] (λ y, real.exp (-(y^2 / 2))) x) / real.exp (-(x^2 / 2))
| 0       _  := by simp [← real.exp_sub]
| (_ + 1) x :=
begin
  let gaussian : ℝ → ℝ := λ x, real.exp (-(x^2 / 2)),
  have gaussian_ne_zero : ∀ x, gaussian x ≠ 0 := by simp [gaussian, real.exp_ne_zero],
  have cont_diff_gaussian : cont_diff ℝ ⊤ gaussian := ((cont_diff_id.pow 2).div_const 2).neg.exp,
  have : ∀ x n,
    (-1 : ℝ)^(n+1) * (deriv^[n+1] gaussian x) / gaussian x
    = x * ((-1 : ℝ)^n * (deriv^[n] gaussian x)) / gaussian x -
      deriv (λ z, (-1 : ℝ)^n * (deriv^[n] gaussian z) / gaussian z) x,
  { intros x n,
    rw [function.iterate_succ_apply', deriv_div _ _ (gaussian_ne_zero _), deriv_const_mul,
        (by simp [gaussian, mul_comm] : deriv gaussian x = (-x) * gaussian x)],
    field_simp [gaussian_ne_zero],
    ring_exp,
    { apply (cont_diff_top_iff_deriv.mp (cont_diff_gaussian.iterate_deriv _)).1, },
    { apply (cont_diff_top_iff_deriv.mp _).1,
      exact (cont_diff_gaussian.iterate_deriv _).const_smul ((-1 : ℝ)^n) },
    { simp } },
  simp_rw [this, hermite_succ, ← mul_div x, ← hermite_eq_deriv_gaussian'],
  simp [aeval_def, eval₂_eq_eval_map],
end
```